spk/transmission/src/service-setup.sh

```diff
@@ -70,17 +70,40 @@ service_postinst ()
 
 service_postupgrade ()
 {
-    # Needed to force correct permissions, during update
     # Extract the right paths from config file
     if [ -r "${CFG_FILE}" ]; then
         if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
+            OLD_DOWNLOAD_DIR=$(sed -n 's/.*"download-dir"\s*:\s*"\(.*\)",/\1/p' "${CFG_FILE}")
+            OLD_INCOMPLETE_DIR=$(sed -n 's/.*"incomplete-dir"\s*:\s*"\(.*\)",/\1/p' "${CFG_FILE}")
+            OLD_WATCHED_DIR=$(sed -n 's/.*"watch-dir"\s*:\s*"\(.*\)",/\1/p' "${CFG_FILE}")
+
+            NEW_DOWNLOAD_DIR="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}"
+            NEW_INCOMPLETE_DIR="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/incomplete"
+            NEW_WATCHED_DIR="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/watch"
+
             # update folder
-            sed -i -e "s|\s\"download-dir\".*|    \"download-dir\": \"${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}\",|g" "${CFG_FILE}"
+            sed -i -e "s|\s\"download-dir\".*|    \"download-dir\": \"${NEW_DOWNLOAD_DIR}\",|g" "${CFG_FILE}"
+            sed -i -e "s|\s\"watch-dir\".*|    \"watch-dir\": \"${NEW_INCOMPLETE_DIR}\",|g" "${CFG_FILE}"
+            sed -i -e "s|\s\"incomplete-dir\".*|    \"incomplete-dir\": \"${NEW_WATCHED_DIR}\",|g" "${CFG_FILE}"
+
+            # move files
+            # not moving download dir because it could contain data not from this package
+            # if [ "$OLD_DOWNLOAD_DIR" != "$NEW_DOWNLOAD_DIR" ]; then
+            #     mv "$OLD_DOWNLOAD_DIR"/* "$NEW_DOWNLOAD_DIR"
+            # fi
+            if [ -n "${OLD_INCOMPLETE_DIR}" ] &&  [ "$OLD_INCOMPLETE_DIR" != "$NEW_INCOMPLETE_DIR" ]; then
+                mkdir -p "$NEW_INCOMPLETE_DIR"
+                mv -nv "$OLD_INCOMPLETE_DIR"/* "$NEW_INCOMPLETE_DIR/"
+            fi
+            if [ -n "${OLD_WATCHED_DIR}" ] && [ "$OLD_WATCHED_DIR" != "$NEW_WATCHED_DIR" ]; then
+                mkdir -p "$NEW_WATCHED_DIR"
+                mv -nv "$OLD_WATCHED_DIR"/* "$NEW_WATCHED_DIR/"
+            fi
```

Comment:
@hgy59 what do you think about the migration? I can't decide whether It's better for the user to do this manually rather than the package.

Reply:
First of all I found a bug in the package.
The new Package is designed to use a "incomplete" folder, located on the new transmission share and be enabled it by default.
The setting for enabling this is not correct, in the setting.json this line is created:

> "incomplete-dir-enabled": @incomplete_dir_enabled@,

But that should be:

> "incomplete-dir-enabled": true,

Secondly I would suggest that the package moves the old *.part file to te new incomplete location.
Where the old .part are located can be found in te settings.json.

If you create a package for a ARM cpu (DS116) I would be happy do do the testing on DSM 7

Reply:
I can not reproduce the `@incomplete_dir_enabled@` issue. Is this from an upgrade or a fresh installation?

The part file migration is exactly the code snip it you're commenting on. So that's a +1 on the package to migrate those, Cool I'm happy with that.

Thanks I'll upload the new build. In the meantime you can always find the latest build at the bottom of a GitHub Action run. The artefacts contain all compiled architectures: https://github.com/SynoCommunity/spksrc/actions/runs/1177151421#artifacts

Reply:
Ok I tested the latest artifact package and it it correct now.
So I tested probably a older version.